### PR TITLE
POC - attr. filter renaming

### PR DESCRIPTION
**Pull request by NestorEncinas**
---

Supported PR commands:

| Command                                               | Description                                                |
| ----------------------------------------------------- | ---------------------------------------------------------- |
| `ok to test`                                          | Re-run standard checks                                     |
| `extended check sonar`                                | SonarQube tests                                            |
| `extended check plugins`                              | Dashboard plugins tests                                    |
| `extended test - backstop`                            | BackstopJS tests                                           |
| **E2E Cypress tests commands - TIGER**                |                                                            |
| `extended test - tiger-cypress - isolated <testName>` | Run isolated tests running against recorded Tiger backend. |
| `extended test - tiger-cypress - record <testName>`   | Create a new recording for isolated Tiger tests.           |
| **E2E Cypress tests commands - BEAR**                 |                                                            |
| `extended test - cypress - isolated <testName>`       | Run isolated tests running against recorded Bear backend.  |
| `extended test - cypress - record <testName>`         | Create a new recording for isolated Bear tests.            |

`<testName>` in cypress commands is used to filter specfiles. Example, to run record with BEAR backend

-   Against `dashboard.spec.ts` and `drilling.spec.ts`, execute command `extended test - cypress - record dashboard,drilling`
-   Against all specfiles, execute command `extended test - cypress - record` or `extended test - cypress - record *`

---

# PR Checklist

-   [ ] commit messages adhere to the [commit message guidelines](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#what-should-the-commits-look-like)
-   [ ] review was done by a Code owner [if necessary](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-tell-if-my-pull-request-needs-approval-by-a-code-owner) (if you think it is not necessary, explain the reasoning in the description or in a comment)
-   [ ] `check` passes
-   [ ] `extended test - backstop` passes
-   [ ] `extended test - tiger-cypress - record` to record new mapping files (Tiger BE)
-   [ ] `extended test - cypress - record` to record new mapping files (Bear BE)
-   [ ] `extended test - tiger-cypress - isolated` passes
-   [ ] `extended test - cypress - isolated` passes
-   [ ] `rush change` [was run if applicable](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-describe-my-changes-for-the-changelog)

### CODE REVIEW

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/11091/ : SUCCESS in 6s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/13185/ : FAILURE in 5m 00s

File: libs/sdk-model/src/dashboard/filterContext.ts [hunk 1]
```diff
@@ -71,6 +71,11 @@ export interface IDashboardAttributeFilter {
          * Parent filters that are limiting elements available in this filter
          */
         filterElementsBy?: IDashboardAttributeFilterParent[];
+
+        /**
+         * Custom title of the filter.
```
Comment (kandl) [inline libs/sdk-model/src/dashboard/filterContext.ts:76]: I think there should be more information about it (e.g., it has priority over the attribute title, etc...)

File: libs/sdk-ui-dashboard/src/model/commands/filters.ts [hunk 2]
```diff
@@ -197,6 +197,49 @@ export function clearDateFilterSelection(correlationId?: string): ChangeDateFilt
...
+/**
+ * @internal
+ */
+export interface SetAttributeFilterTitlePayload {
+    filterLocalId: string;
+    title: string;
+}
+
+/**
+ * @internal
```
Comment (kandl) [inline libs/sdk-ui-dashboard/src/model/commands/filters.ts:209]: why @internal?

File: libs/sdk-ui-dashboard/src/model/commands/filters.ts [hunk 3]
```diff
@@ -472,6 +520,7 @@ export function changeAttributeFilterSelection(
     filterLocalId: string,
     elements: IAttributeElements,
     selectionType: AttributeFilterSelectionType,
+    title?: string,
```
Comment (kandl) [inline libs/sdk-ui-dashboard/src/model/commands/filters.ts:523]: This is problematic and I would  consider it as a "breaking change", as the command is already public. But does it make sense at all to include title in this command? It's mainly about changing the selection? What is the use case to include title here?

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/11098/ : SUCCESS in 6s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/13192/ : FAILURE in 5m 38s

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/11132/ : SUCCESS in 11s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/13232/ : FAILURE in 14m 29s

**Pull request closed without merging**